primitives/core/src/offchain/testing.rs

```diff
@@ -160,11 +158,21 @@ impl offchain::Externalities for TestOffchainExt {
 	}
 
 	fn timestamp(&mut self) -> Timestamp {
-		Timestamp::from_unix_millis(self.0.read().timestamp)
+		use std::convert::TryInto;
+		let millis = std::time::SystemTime::now()
```

Comment:
I was considering that too, but that would not work for concurrent ocw tests i.e.. For the above case  I was think of an additional `fn` allowing to advance the current timestamp by a duration.

Reply:
How so? Every test should have it's own externalities anyway, the state is not shared between threads or tests at all.

Reply:
I meant multiple threads to test concurrency within one test, that was my angle/context.

Reply:
I'm not sure that test make much sense - if you don't have control over how the threads progress how do you make sure that you actually test concurrent access? The failure of such test would be super spurious, so I don't think it's even worth to consider.

Reply:
Coming from a different angle, the actual concurrency test here would test atomicity of `fn mutate(..)`, since that is the actual atomic operation, so the test really would not test anything implemented in this PR.
The lock state is already being tested as part of a test.